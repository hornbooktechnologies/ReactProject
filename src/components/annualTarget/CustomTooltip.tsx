 
  import { ChartConstant, DateformatConstant } from "../../utils/AppConstants";
  import { CustomTooltipModel } from "../../models/LinechartModel";
  import Moment from 'moment'
import { cssClassName } from "../../utils/CssConstants";

  //Custom Dots for Tooltip
  const CustomDot = ({ fill} : string|any) => {
    return (
      <>
      <svg className="svg">
          <circle cx={9} cy={9} r={6} fill={fill} />
        </svg>
      </>
    );
  };

  const CustomTooltip = ({ selectValue,linechartData ,active, payload }: CustomTooltipModel) => {
    if (active && payload !== undefined && payload && payload.length) {
      payload = payload.filter((payloadData) => {
        return !payloadData.name.includes("hide");
      });
    }

    if (active && payload && payload.length) {
      return (
       
        <div className="rechart-custom-tooltip annualtarget-tooltip">
          {selectValue === ChartConstant.DAILY_DATA ? (
            <div className={`${cssClassName.TOOLTIP_PERIOD_SECTION}`}>{`${Moment(
              payload[0].payload.Period
            ).format(DateformatConstant.YYY_MMM_DD_HH_mm)}`}</div>
          ) : selectValue === ChartConstant.ANNUAL_DATA ? (
            <div className={`${cssClassName.TOOLTIP_PERIOD_SECTION}`}>{`${Moment(
              payload[0].payload.Period
            ).format(DateformatConstant.YYY_MMM)}`}</div>
          ) : (
            <div className={`${cssClassName.TOOLTIP_PERIOD_SECTION}`}>{`${Moment(
              payload[0].payload.Period
            ).format(DateformatConstant.YYY_MMM_DD)}`}</div>
          )}
          <>
            <ul className="tooltip-main-section">
              {linechartData.Prefix !== null ? (
                <>
                  <li className={`${cssClassName.TOOLTIP_NAME_SECTION}`}>
                    <CustomDot
                      fill={
                        payload[0].name === ChartConstant.FORECAST
                          ? ChartConstant.FORECAST_COLOR
                          : payload[0].name === ChartConstant.PLANNED
                          ? ChartConstant.PLANNED_COLOR
                          : ChartConstant.ACTUAL_COLOR
                      }
                    />
                    {`${payload[0].name}`}:
                    {` ${linechartData.Prefix} ${payload[0].value} ${linechartData.Suffix} `}
                  </li>
                </>
              ) : (
                <>
                  <li className={`${cssClassName.TOOLTIP_NAME_SECTION}`}>
                    <CustomDot
                      fill={
                        payload[0].name === ChartConstant.FORECAST
                          ? ChartConstant.FORECAST_COLOR
                          : payload[0].name === ChartConstant.PLANNED
                          ? ChartConstant.PLANNED_COLOR
                          : ChartConstant.ACTUAL_COLOR
                      }
                    />
                    {`${payload[0].name}`}:
                    { `${payload[0].value} ${linechartData.Suffix} `}
                  </li>
                </>
              )}

              {payload.length > 1 ? (
                <>
                  {linechartData.Prefix !== null ? (
                    <>
                      <li className={`${cssClassName.TOOLTIP_NAME_SECTION}`}>
                        <CustomDot
                          fill={
                            payload[1].name === ChartConstant.FORECAST
                              ? ChartConstant.FORECAST_COLOR
                              : payload[1].name === ChartConstant.PLANNED
                              ? ChartConstant.PLANNED_COLOR
                              : ChartConstant.ACTUAL_COLOR
                          }
                        />
                        {`${payload[1].name}`}:
                        { ` ${linechartData.Prefix} ${payload[1].value} ${linechartData.Suffix}`}
                      </li>
                    </>
                  ) : (
                    <>
                      <li className={`${cssClassName.TOOLTIP_NAME_SECTION}`}>
                        <CustomDot
                          fill={
                            payload[1].name === ChartConstant.FORECAST
                              ? ChartConstant.FORECAST_COLOR
                              : payload[1].name === ChartConstant.PLANNED
                              ? ChartConstant.PLANNED_COLOR
                              : ChartConstant.ACTUAL_COLOR
                          }
                        />
                        {`${payload[1].name}`}:
                        {` ${payload[1].value} ${linechartData.Suffix}`}
                      </li>
                    </>
                  )}
                </>
              ) : null}

              {payload.length > 2 ? (
                <>
                  {linechartData.Prefix !== null ? (
                    <>
                      <li className={`${cssClassName.TOOLTIP_NAME_SECTION}`}>
                        <CustomDot
                          fill={
                            payload[2].name === ChartConstant.FORECAST
                              ? ChartConstant.FORECAST_COLOR
                              : payload[2].name === ChartConstant.PLANNED
                              ? ChartConstant.PLANNED_COLOR
                              : ChartConstant.ACTUAL_COLOR
                          }
                        />
                        {`${payload[2].name}`}:
                        {` ${linechartData.Prefix}${payload[2].value} ${linechartData.Suffix}`}
                      </li>
                    </>
                  ) : (
                    <>
                      <li className={`${cssClassName.TOOLTIP_NAME_SECTION}`}>
                        <CustomDot
                          fill={
                            payload[2].name === ChartConstant.FORECAST
                              ? ChartConstant.FORECAST_COLOR
                              : payload[2].name === ChartConstant.PLANNED
                              ? ChartConstant.PLANNED_COLOR
                              : ChartConstant.ACTUAL_COLOR
                          }
                        />
                        {`${payload[2].name}`}:
                        {` ${payload[2].value} ${linechartData.Suffix}`}
                      </li>
                    </>
                  )}
                </>
              ) : null}
            </ul>
          </>
        </div>
      );
    }

    return null;
  };
  export default CustomTooltip;