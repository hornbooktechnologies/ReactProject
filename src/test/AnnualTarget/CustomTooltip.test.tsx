import React from 'react'
// import CustomTooltip from '../../components/common/lineChart/CustomTooltip';
import CustomTooltip from '../../components/annualTarget/CustomTooltip'
import renderer from 'react-test-renderer'
describe('Component: CustomTooltip', () => {
  it('CustomTooltip renders correctly', () => {
    var selectValue = 'AnnualData'
    var linechartData = {
      currentDisable: false,
      graphData:
        '{"Actua:[52.7,43.41,45.19,64.06,55.43,51.66,217.55,374.96,311.73,359.21,362.63,null],"Perio:["2021-04-01T00:00","2021-05-01T00:00","2021-06-01T00:00","2021-07-01T00:00","2021-08-01T00:00","2021-09-01T00:00","2021-10-01T00:00","2021-11-01T00:00","2021-12-01T00:00","2022-01-01T00:00","2022-02-01T00:00","2022-03-01T00:00"],"Planne:[15.31,18.49,49.32,68.43,-5.67,43.35,74.05,76.23,102.75,108.93,88.2,66.98],"Forecas:[null,null,null,null,null,null,null,null,null,null,null,438.43],"Estimate:[690.37,2078.44,206.54]}',
      indexFound: true,
      nextDisable: false,
      datelabel: '2021',
      Prefix: 'Â¥',
      Suffix: 'Oku',
      Suffix2: 'Oku',
    }
    var active = false
    var payload = [
      {
        stroke: '#ED9933',
        strokeWidth: '3',
        fill: '#fff',
        points: [],
        dataKey: 'Planned',
        name: 'Planned',
        color: '#ED9933',
        value: 15.31,
        payload: {
          Period: '2021-04-01T00:00',
          Forecast: null,
          Actual: 52.7,
          Planned: 15.31,
          TimeStamp: 1617215400000,
        },
      },
      {
        stroke: '#E0515A',
        strokeWidth: '3',
        fill: '#fff',
        points: [],
        dataKey: 'Actual',
        name: 'Actual',
        color: '#E0515A',
        value: 52.7,
        payload: {
          Period: '2021-04-01T00:00',
          Forecast: null,
          Actual: 52.7,
          Planned: 15.31,
          TimeStamp: 1617215400000,
        },
      },
    ]
    const tree = renderer
      .create(
        <CustomTooltip
          selectValue={selectValue}
          linechartData={linechartData}
          active={active}
          payload={payload}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
