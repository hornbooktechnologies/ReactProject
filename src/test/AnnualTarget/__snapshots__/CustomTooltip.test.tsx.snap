// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CustomTooltip CustomTooltip renders correctly 1`] = `null`;
